libcontainer/cgroups/systemd/v1.go

```diff
@@ -355,13 +356,20 @@ func (m *legacyManager) freezeBeforeSet(unitName string, r *configs.Resources) (
 		//
 		// Do not return errors from getUnitProperty, as they alone
 		// should not prevent Set from working.
-		devPolicy, e := getUnitProperty(m.dbus, unitName, "DevicePolicy")
+		devPolicy, e := getUnitTypeProperty(m.dbus, unitName, getUnitType(unitName), "DevicePolicy")
 		if e == nil && devPolicy.Value == dbus.MakeVariant("auto") {
-			devAllow, e := getUnitProperty(m.dbus, unitName, "DeviceAllow")
-			if e == nil && devAllow.Value == dbus.MakeVariant([]deviceAllowEntry{}) {
-				needsFreeze = false
-				needsThaw = false
-				return
+			devAllow, e := getUnitTypeProperty(m.dbus, unitName, getUnitType(unitName), "DeviceAllow")
+			if e == nil {
+				rv := reflect.ValueOf(devAllow.Value.Value())
```

Comment:
Do you mean that `devAllow.Value == dbus.MakeVariant([]deviceAllowEntry{})` did not work? Or that your version is more readable? (or both?)

Reply:
> Do you mean that devAllow.Value == dbus.MakeVariant([]deviceAllowEntry{}) did not work? Or that your version is more readable? (or both?)

Well, it doesn't seen to work for me. Does that for you, in that case there is something wrong with my setup? :thinking:

Reply:
I think I improved the readability a notch in https://github.com/opencontainers/runc/pull/3148:

```go
                        devAllow, e := getUnitTypeProperty(m.dbus, unitName, unitType, "DeviceAllow")
                          if e == nil {
                                  if rv := reflect.ValueOf(devAllow.Value.Value()); rv.Kind() == reflect.Slice && rv.Len() == 0 {
                                          needsFreeze = false
                                          needsThaw = false
                                          return
                                  }
                          }
```

Reply:
Guess I can do that instead if the others agree, even tho. the line is a bit too long. I'll update the PR.

Reply:
What about this:
```go
                        if e == nil { 
                                rv := reflect.ValueOf(devAllow.Value.Value())
                                if rv.Kind() == reflect.Slice && rv.Len() == 0 { 
```

Reply:
To be it reads slightly better than the switch/case (although I agree that either version we came with so far is not super readable).